src/x11.rs

```diff
@@ -758,6 +767,45 @@ impl Drop for X11Impl {
     }
 }
 
+/// Create a shared memory identifier.
+fn create_shm_id() -> io::Result<fd::OwnedFd> {
```

Comment:
We should move the `create_memfile` logic in the `wayland` backend to somewhere it can be used in both backends, instead of duplicating things.

Using `memfd_create` is better than `shm_open` where supported.

Reply:
I'm not sure if memfiles are supported for X11 MIT-SHM

Reply:
I can't really find any documentation for `ShmAttachFd` beyond:

> Create a shared memory segment.  The file descriptor will be mapped at offset
zero, and the size will be obtained using fstat().  A zero size will result in a
Value error.

> The file descriptor the server should mmap().

So any fd it can `fstat` to get the size and `mmap` should be fine.

https://lists.freedesktop.org/archives/xorg-devel/2013-October/038490.html

> I was hoping for a patch to shmproto.txt, but there isn't one, le sigh.
We should really fix the MIT-SHM docs, Jon's reverse-engineered spec pdf
is full of enough handwave to be embarassing for us.

Lol. I guess the fact https://gitlab.freedesktop.org/xorg/proto/xcbproto/-/blob/master/src/shm.xml exists means the blocker there is addressed, but no one has added documentation for version 1.2 of the protocol yet.